server/instrumentation-backend/src/main/java/sh/calaba/instrumentationbackend/actions/text/TextAction.java

```diff
@@ -63,6 +66,36 @@ public Result call() throws Exception {
         }
     }
 
+    /*
+     * On Android 4 devices we have servedView instanceof WebView and correct input connection, so we don't need to use js logic on old devices.
+     */
+    public static boolean requiresWebViewInput(View view) {
+        return view instanceof WebView && Build.VERSION.SDK_INT > 27;
+    }
+
+    /*
+     * Executes JS to handle WebView input operations.
+     */
+    public static Result evalWebViewInputScript(WebView webView, String script, Object ... scriptParams) {
+        try {
+            // Execute JS on the UI thread
+            webView.post(new Runnable() {
+                @Override
+                public void run() {
+                    WebSettings webSettings = webView.getSettings();
+                    webSettings.setJavaScriptEnabled(true);
+
+                    webView.evaluateJavascript(String.format(script, scriptParams), null);
+                }
+            });
+
+            return Result.successResult();
```

Comment:
What happens if the posted runnable throws? Seems we'd always return success... Is there any way to have `webView.post()` return a Future we can block on so we have synchronous operation? Or are we dependent on have async evaluation of the input?

Reply:
Technically we can use `@JavascriptInterface` and two methods (success and fail), but anyway this is async operations. I will try to use  FutureTask then